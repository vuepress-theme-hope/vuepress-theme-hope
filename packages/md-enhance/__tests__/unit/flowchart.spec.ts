import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { flowchart } from "../../src/node/markdown-it/index.js";

const demo = `st=>start: Start|past:>http://www.google.com[blank]
e=>end: End|future:>http://www.google.com
op1=>operation: My Operation|past
op2=>operation: Stuff|current
sub1=>subroutine: My Subroutine|invalid
cond=>condition: Yes
or No?|approved:>http://www.google.com
c2=>condition: Good idea|rejected
io=>inputoutput: catch something...|future

st->op1(right)->cond
cond(yes, right)->c2
cond(no)->sub1(left)->op1
c2(yes)->io->e
c2(no)->op2->e`;

describe("flowchart", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(flowchart);

  it("Should render ```flow", () => {
    const flowRenderResult = markdownIt.render(`
\`\`\`flow
${demo}
\`\`\`
`);

    expect(flowRenderResult).toMatch(
      /<FlowChart id="flowchart-.*?" code=".*?" preset="vue"><\/FlowChart>/
    );
    expect(flowRenderResult).toMatchSnapshot();
  });

  it("Should render ```flowchart", () => {
    const flowChartRenderResult = markdownIt.render(`
\`\`\`flowchart
${demo}
\`\`\`
`);

    expect(flowChartRenderResult).toMatch(
      /<FlowChart id="flowchart-.*?" code=".*?" preset="vue"><\/FlowChart>/
    );
    expect(flowChartRenderResult).toMatchSnapshot();
  });

  it("Should not render", () => {
    expect(
      markdownIt.render(`
${demo}
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
\`\`\`md
${demo}
\`\`\`
`)
    ).toMatchSnapshot();
  });

  it("Should handle preset", () => {
    const flowRenderResult = markdownIt.render(`
\`\`\`flow:ant
${demo}
\`\`\`
`);

    expect(flowRenderResult).toMatch(
      /<FlowChart id="flowchart-.*?" code=".*?" preset="ant"><\/FlowChart>/
    );
    expect(flowRenderResult).toMatchSnapshot();

    const flowChartRenderResult = markdownIt.render(`
\`\`\`flowchart:ant
${demo}
\`\`\`
`);

    expect(flowChartRenderResult).toMatch(
      /<FlowChart id="flowchart-.*?" code=".*?" preset="ant"><\/FlowChart>/
    );
    expect(flowChartRenderResult).toMatchSnapshot();
  });
});
