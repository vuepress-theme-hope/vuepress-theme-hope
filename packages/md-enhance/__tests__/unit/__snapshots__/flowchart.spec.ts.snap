// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flowchart > Should handle preset 1`] = `"<FlowChart id=\\"flowchart-0\\" code=\\"eJx1ULFOwzAQ3f0VHotEUrWjpZgJMQFDJoQYXPuSGFKfZZ8bVfLHY6dQwdDlrHd+7927i9TJSCqQ4H19sleRhJyIvNhul2VpR8Rxhlbj8f0wK/f1waCT4Izgj87kIVEKcEPA0O86iR6CIotO8Oczf/1F66TC2P9j9JSGIesUAjhiMR2KQakBE1kHq0N/hdm6k5qtYRqd6WSt9mLzBpFh4C/4kJX3AU9gbmXUJcAf5ROi4daAygE+QRMYZrGT1vlEZWqpgmtFeuIRj0CTdWPbtj93YCxSU9bZbYIdJ7prVuc13uYM8Z5f2/tL02EBdcvNDEP9KNqSqJILsNhIqHCllVNV+A2yapmT\\" preset=\\"ant\\"></FlowChart>"`;

exports[`flowchart > Should handle preset 2`] = `"<FlowChart id=\\"flowchart-0\\" code=\\"eJx1ULFOwzAQ3f0VHotEUrWjpZgJMQFDJoQYXPuSGFKfZZ8bVfLHY6dQwdDlrHd+7927i9TJSCqQ4H19sleRhJyIvNhul2VpR8Rxhlbj8f0wK/f1waCT4Izgj87kIVEKcEPA0O86iR6CIotO8Oczf/1F66TC2P9j9JSGIesUAjhiMR2KQakBE1kHq0N/hdm6k5qtYRqd6WSt9mLzBpFh4C/4kJX3AU9gbmXUJcAf5ROi4daAygE+QRMYZrGT1vlEZWqpgmtFeuIRj0CTdWPbtj93YCxSU9bZbYIdJ7prVuc13uYM8Z5f2/tL02EBdcvNDEP9KNqSqJILsNhIqHCllVNV+A2yapmT\\" preset=\\"ant\\"></FlowChart>"`;

exports[`flowchart > Should not render 1`] = `
"<p>st=&gt;start: Start|past:&gt;<a href=\\"http://www.google.com\\">http://www.google.com</a>[blank]
e=&gt;end: End|future:&gt;<a href=\\"http://www.google.com\\">http://www.google.com</a>
op1=&gt;operation: My Operation|past
op2=&gt;operation: Stuff|current
sub1=&gt;subroutine: My Subroutine|invalid
cond=&gt;condition: Yes
or No?|approved:&gt;<a href=\\"http://www.google.com\\">http://www.google.com</a>
c2=&gt;condition: Good idea|rejected
io=&gt;inputoutput: catch something...|future</p>
<p>st-&gt;op1(right)-&gt;cond
cond(yes, right)-&gt;c2
cond(no)-&gt;sub1(left)-&gt;op1
c2(yes)-&gt;io-&gt;e
c2(no)-&gt;op2-&gt;e</p>
"
`;

exports[`flowchart > Should not render 2`] = `
"<pre><code class=\\"language-md\\">st=&gt;start: Start|past:&gt;http://www.google.com[blank]
e=&gt;end: End|future:&gt;http://www.google.com
op1=&gt;operation: My Operation|past
op2=&gt;operation: Stuff|current
sub1=&gt;subroutine: My Subroutine|invalid
cond=&gt;condition: Yes
or No?|approved:&gt;http://www.google.com
c2=&gt;condition: Good idea|rejected
io=&gt;inputoutput: catch something...|future

st-&gt;op1(right)-&gt;cond
cond(yes, right)-&gt;c2
cond(no)-&gt;sub1(left)-&gt;op1
c2(yes)-&gt;io-&gt;e
c2(no)-&gt;op2-&gt;e
</code></pre>
"
`;

exports[`flowchart > Should render \`\`\`flow 1`] = `"<FlowChart id=\\"flowchart-0\\" code=\\"eJx1ULFOwzAQ3f0VHotEUrWjpZgJMQFDJoQYXPuSGFKfZZ8bVfLHY6dQwdDlrHd+7927i9TJSCqQ4H19sleRhJyIvNhul2VpR8Rxhlbj8f0wK/f1waCT4Izgj87kIVEKcEPA0O86iR6CIotO8Oczf/1F66TC2P9j9JSGIesUAjhiMR2KQakBE1kHq0N/hdm6k5qtYRqd6WSt9mLzBpFh4C/4kJX3AU9gbmXUJcAf5ROi4daAygE+QRMYZrGT1vlEZWqpgmtFeuIRj0CTdWPbtj93YCxSU9bZbYIdJ7prVuc13uYM8Z5f2/tL02EBdcvNDEP9KNqSqJILsNhIqHCllVNV+A2yapmT\\" preset=\\"vue\\"></FlowChart>"`;

exports[`flowchart > Should render \`\`\`flowchart 1`] = `"<FlowChart id=\\"flowchart-0\\" code=\\"eJx1ULFOwzAQ3f0VHotEUrWjpZgJMQFDJoQYXPuSGFKfZZ8bVfLHY6dQwdDlrHd+7927i9TJSCqQ4H19sleRhJyIvNhul2VpR8Rxhlbj8f0wK/f1waCT4Izgj87kIVEKcEPA0O86iR6CIotO8Oczf/1F66TC2P9j9JSGIesUAjhiMR2KQakBE1kHq0N/hdm6k5qtYRqd6WSt9mLzBpFh4C/4kJX3AU9gbmXUJcAf5ROi4daAygE+QRMYZrGT1vlEZWqpgmtFeuIRj0CTdWPbtj93YCxSU9bZbYIdJ7prVuc13uYM8Z5f2/tL02EBdcvNDEP9KNqSqJILsNhIqHCllVNV+A2yapmT\\" preset=\\"vue\\"></FlowChart>"`;
